sdk/tests/conformance/glsl/bugs/bool-type-cast-bug.html

```diff
@@ -0,0 +1,317 @@
+
+<script id="vshader-float2" type="x-shader/x-vertex">
+attribute vec3 aPosition;
+
+void main() {
+    gl_Position = vec4(aPosition, 1);
+}
+</script>
+<script id="fshader-float2" type="x-shader/x-fragment">
+uniform bool bvalue;
+
+void main() {
+    mediump float fvalue = float(bvalue);
+
+    if (fvalue == 1.0)
+        gl_FragColor = vec4(1.0, 0.0, 0.0, 1.0);
+    else if (fvalue == 0.0)
+        gl_FragColor = vec4(0.0, 1.0, 0.0, 1.0);
+    else
+        gl_FragColor = vec4(0.0, 0.0, 0.0, 1.0);
+}
+</script>
+
+<script id="vshader-uint1" type="x-shader/x-vertex">#version 300 es
+in vec3 aPosition;
+flat out uint uvalue;
+uniform bool bvalue;
+
+void main() {
+    gl_Position = vec4(aPosition, 1);
+    uvalue = uint(bvalue);
+}
+</script>
+<script id="fshader-uint1" type="x-shader/x-fragment">#version 300 es
+flat in uint uvalue;
+out highp vec4 myFragColor;
+
+void main() {
+    if (uvalue == 1u)
+        myFragColor = vec4(1.0, 0.0, 0.0, 1.0);
+    else if (uvalue == 0u)
+        myFragColor = vec4(0.0, 1.0, 0.0, 1.0);
+    else
+        myFragColor = vec4(0.0, 0.0, 0.0, 1.0);
+}
+</script>
+
+<script id="vshader-uint2" type="x-shader/x-vertex">#version 300 es
+in vec3 aPosition;
+
+void main() {
+    gl_Position = vec4(aPosition, 1);
+}
+</script>
+<script id="fshader-uint2" type="x-shader/x-fragment">#version 300 es
+uniform bool bvalue;
+out highp vec4 myFragColor;
+
+void main() {
+    uint uvalue = uint(bvalue);
+
+    if (uvalue == 1u)
+        myFragColor = vec4(1.0, 0.0, 0.0, 1.0);
+    else if (uvalue == 0u)
+        myFragColor = vec4(0.0, 1.0, 0.0, 1.0);
+    else
+        myFragColor = vec4(0.0, 0.0, 0.0, 1.0);
+}
+</script>
+
+<script type="application/javascript">
+"use strict";
+description("Verify int(bool) float(bool) uint(bool) works correctly");
+debug("");
+var wtu = WebGLTestUtils;
+function test() {
+    var gl = wtu.create3DContext("canvas");
+    if (!gl) {
+        testFailed("context does not exist");
+        return;
+    }
+    wtu.setupUnitQuad(gl);
+
+    debug("");
+    var program = wtu.setupProgram(gl, ["vshader-int1", "fshader-int1"], ["aPosition"]);
+    if (!program) {
+        testFailed("Fail to set up program");
+    } else {
+        var uniformLoc = gl.getUniformLocation(program, 'bvalue');
+        debug("Testing vertex shader int(false)");
+        gl.uniform1i(uniformLoc, 0);
+        wtu.drawUnitQuad(gl);
+        wtu.checkCanvas(gl, [0, 255, 0, 255]);
+        debug("Testing vertex shader int(true)");
+        gl.uniform1i(uniformLoc, 1);
+        wtu.drawUnitQuad(gl);
+        wtu.checkCanvas(gl, [255, 0, 0, 255]);
+        gl.deleteProgram(program);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Should be no errors from testing");
+    }
+
+    debug("");
+    program = wtu.setupProgram(gl, ["vshader-int2", "fshader-int2"], ["aPosition"]);
+    if (!program) {
+        testFailed("Fail to set up program");
+    } else {
+        var uniformLoc = gl.getUniformLocation(program, 'bvalue');
```

Comment:
JSHint has trained me against redefinition of variables in the same scope (`var uniformLoc; var uniformLoc;`). I don't think it really matters, I think it's mainly to guard against accidental variable collisions.